src/compo/icartt_nc2ioda.py

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/env python3
+
+#
+# (C) Copyright 2024 UCAR
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+
+import sys
+import argparse
+import netCDF4 as nc
+import numpy as np
+import pandas as pd
+from datetime import datetime, timedelta
+import os
+from pathlib import Path
+
+import xarray as xr
+import math
+from numpy import log as ln
+
+import pyiodaconv.ioda_conv_engines as iconv
+from collections import defaultdict, OrderedDict
+from pyiodaconv.orddicts import DefaultOrderedDict
+
+
+# constants
+HPA2PA = 1E2
+ppbv2molmol = 1e-9
+pptv2molmol = 1e-12
+
+# {'iodaName' : ['obsName', 'iodaUnit', 'obsToIodaUnivConv']
+obsvars = {'nitrogendioxideInsitu': ['NO2_ACES', 'mol mol-1', ppbv2molmol],
+           'carbonmonoxideInsitu': ['CO_ppb', 'mol mol-1', ppbv2molmol],
+           'ozoneInsitu': ['O3_CL', 'mol mole-1', ppbv2molmol],
+           'formaldehydeInsitu': ['CH2O_ISAF', 'mol mol-1', pptv2molmol],
+           'airTemperature': ['T', 'K', 1],
+           'windEastward': ['U', 'm s-1', 1],
+           'windNorthward': ['V', 'm s-1', 1]}
+
+
+class icartt(object):
+
+    def __init__(self, filenames):
+
+        self.filenames = filenames
+        self.make_dictionaries()      # Set up variable names for IODA
+        self.DimDict = {}
+        self.read()    # Read data from file
+
+    def read(self):
+
+        # Loop through input filenames
+        first = True
+        for filename in self.filenames:
+            try:
+                dsFlight = xr.open_dataset(filename)
```

Comment:
this is so nice @mer-a-o ,,, been looking for something more like this

thanks